public/pages/Findings/components/FindingDetailsFlyout.tsx

```diff
@@ -121,12 +122,21 @@ export default class FindingDetailsFlyout extends Component<
     };
   }
 
+  componentWillUnmount(): void {
+    this.abortGetFindingsSignals.forEach(abort => {
+      abort.signal = true;
+    })
+    this.abortGetFindingsSignals = [];
+  }
+
   getCorrelations = async () => {
     const { id, detector } = this.props.finding;
     let allFindings = this.props.findings;
     if (this.props.shouldLoadAllFindings) {
       // if findings come from the alerts fly-out, we need to get all the findings to match those with the correlations
-      allFindings = await DataStore.findings.getAllFindings();
+      const abort = { signal: false };
```

Comment:
not a blocker but just curious, why we are not using AbortController class, which provides a more standard way of handling cancellation signals in JavaScript?

```
const abortController = new AbortController();
const abortSignal = abortController.signal;

this.abortGetFindingsSignals.push(abortController);
allFindings = await DataStore.findings.getAllFindings({ signal: abortSignal });
```
// Later, if you need to abort the operation:
// abortController.abort();

Reply:
We could use the in built AbortController too, essentially the same pattern.